babel/messages/pofile.py

```diff
@@ -313,15 +322,17 @@ def parse(self, fileobj):
             self.translations.append([0, _NormalizedString(u'""')])
             self._add_message()
 
-    def _invalid_pofile(self, line, lineno, msg):
+    def _invalid_pofile(self, line, lineno, msg) -> None:
         assert isinstance(line, str)
         if self.abort_invalid:
             raise PoFileError(msg, self.catalog, line, lineno)
         print("WARNING:", msg)
         print(f"WARNING: Problem on line {lineno + 1}: {line!r}")
 
 
-def read_po(fileobj, locale=None, domain=None, ignore_obsolete=False, charset=None, abort_invalid=False):
+def read_po(fileobj: Iterable[str | bytes], locale: str | Locale | None = None,
```

Comment:
Is `Iterable[str | bytes]` actually correct here? It looks like it should be a filelike.

Reply:
What type do you suggest? `IO[bytes] | IO[str]` seems to work.

Reply:
I think [`IO[AnyStr]` is the usual idiom](https://docs.python.org/3.11/library/typing.html#typing.IO)?